File: src_local/agents/Support/memory_persistence.py
import json
import logging
from pathlib import Path

logger = logging.getLogger(__name__)

class MemoryPersistence:
    """Assistente Técnico: Guardião da Memória e Persistência de Dados 💾"""
    
    def __init__(self, storage_path: Path):
        self.path = storage_path

    def load_ledger(self) -> dict:
        """Carrega a memória de longo prazo com segurança."""
        if self.path.exists():
            try:
                return json.loads(self.path.read_text(encoding='utf-8'))
            except Exception as e:
                logger.error(f"Falha ao carregar ledger: {e}")
                return {}
        return {}

    def save_ledger(self, data: dict):
        """Persiste a memória de forma atômica."""
        try:
            self.path.parent.mkdir(parents=True, exist_ok=True)
            self.path.write_text(json.dumps(data, indent=2), encoding='utf-8')
        except Exception as e:
            logger.error(f"Falha ao salvar ledger: {e}")

    def get_file_metadata(self, ledger: dict, file_path: str) -> dict:
        """Busca histórico específico de um arquivo."""
        return ledger.get(str(file_path), {})

File: src_local/agents/Support/Core/memory_persistence.ts
import winston from "winston";
import { Path } from "../../../core/path_utils.ts";
import { existsSync, readFileSync, writeFileSync, mkdirSync } from "node:fs";

const logger = winston.child({ module: "MemoryPersistence" });

/**
 * Assistente Técnico: Guardião da Memória e Persistência de Dados 💾 (Bun Version).
 */
export class MemoryPersistence {
    private path: Path;

    constructor(storagePath: Path) {
        this.path = storagePath;
    }

    loadLedger(): Record<string, any> {
        /** Carrega a memória de longo prazo com segurança. */
        if (existsSync(this.path.toString())) {
            try {
                const content = readFileSync(this.path.toString(), "utf-8");
                return JSON.parse(content);
            } catch (e) {
                logger.error(`Falha ao carregar ledger: ${e}`);
                return {};
            }
        }
        return {};
    }

    saveLedger(data: Record<string, any>) {
        /** Persiste a memória de forma atômica. */
        try {
            const parentDir = this.path.parent().toString();
            if (!existsSync(parentDir)) {
                mkdirSync(parentDir, { recursive: true });
            }
            writeFileSync(this.path.toString(), JSON.stringify(data, null, 2), "utf-8");
        } catch (e) {
            logger.error(`Falha ao salvar ledger: ${e}`);
        }
    }

    getFileMetadata(ledger: Record<string, any>, filePath: string): any {
        /** Busca histórico específico de um arquivo. */
        return ledger[filePath] || {};
    }
}
